### PR TITLE
Con 1082 all reviews are visible on extra review page

### PR DESCRIPTION
**Pull request by sznowicki**
Require pwa-common-commerce CON-1082 branch changes.

### CODE REVIEW

File: components/Reviews/components/AllReviewsLink/style.js [hunk 1]
```diff
@@ -9,11 +9,23 @@ import { css } from 'glamor';
 import colors from 'Styles/colors';
 import variables from 'Styles/variables';
 
-export default css({
+/**
+ * Styles for a link parent container.
+ * @type {string}
+ */
+export const containerStyle = css({
```
Comment (richardgorman) [inline components/Reviews/components/AllReviewsLink/style.js]: No named exports in styles please.

File: components/Reviews/components/AllReviewsLink/connector.js [hunk 2]
```diff
@@ -15,6 +18,7 @@ import { getProductReviewCount } from '@shopgate/pwa-common-commerce/product/sel
  */
 const mapStateToProps = state => ({
   totalReviewCount: getProductReviewCount(state),
+  productId: getCurrentProductId(state),
```
Comment (richardgorman) [inline components/Reviews/components/AllReviewsLink/connector.js]: Alphabetical order please

File: components/Reviews/components/AllReviewsLink/index.jsx [hunk 3]
```diff
@@ -7,30 +7,38 @@
...
 import connect from './connector';
-import styles from './style';
+import {
+  containerStyle,
+  linkStyle,
+} from './style';
 
 /**
  * Link to all reviews.
+ * @param {string} productId ProductId.
```
Comment (richardgorman) [inline components/Reviews/components/AllReviewsLink/index.jsx]: `The product id`.

File: components/Reviews/components/AllReviewsLink/index.jsx [hunk 4]
```diff
@@ -7,30 +7,38 @@
...
  * @returns {JSX|null}
  */
-const AllReviewsLink = ({ totalReviewCount }) => {
-  if (totalReviewCount < REVIEW_PREVIEW_LIMIT) {
+const AllReviewsLink = ({ productId, totalReviewCount }) => {
+  if (totalReviewCount <= REVIEW_PREVIEW_LIMIT) {
     return null;
   }
-
+  const publicProductId = bin2hex(productId);
```
Comment (richardgorman) [inline components/Reviews/components/AllReviewsLink/index.jsx]: Line breaks missing before and after this line.

File: components/Reviews/components/AllReviewsLink/index.jsx [hunk 5]
```diff
@@ -7,30 +7,38 @@
...
-import styles from './style';
+import {
+  containerStyle,
+  linkStyle,
+} from './style';
 
 /**
  * Link to all reviews.
+ * @param {string} productId ProductId.
  * @param {number} totalReviewCount The number of reviews given.
```
Comment (richardgorman) [inline components/Reviews/components/AllReviewsLink/index.jsx]: Param description is incorrect. Is this not the total number of reviews that a product has?

File: components/Reviews/components/AllReviewsLink/style.js (hunk 1 above)
Comment (richardgorman) [reply components/Reviews/components/AllReviewsLink/style.js]: Also please don't prefix everything with `Style`.

File: components/Reviews/components/Header/components/RatingCount/spec.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React from 'react';
```
Comment (richardgorman) [inline components/Reviews/components/Header/components/RatingCount/spec.jsx]: Space before LICENCE block.

File: components/Reviews/mock.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const mockedProduct = {
```
Comment (richardgorman) [inline components/Reviews/mock.js]: Space before LICENCE block.

File: components/Reviews/mock.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    count: 4,
+  },
+};
+
+/**
+ * Creates mocked review object.
+ * @param {string|number} id Anything that can be an id.
+ * @return {Object}
+ */
+export const makeReview = id => ({
```
Comment (richardgorman) [inline components/Reviews/mock.js]: `mockReview`?

File: components/Reviews/spec.jsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React from 'react';
```
Comment (richardgorman) [inline components/Reviews/spec.jsx]: Space before LICENCE block.

File: components/View/mock.jsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { Component } from 'react';
```
Comment (richardgorman) [inline components/View/mock.jsx]: Space before LICENCE block.

File: pages/Product/constants.js [hunk 11]
```diff
@@ -8,12 +8,11 @@
...
-
 export const OPEN_PRODUCT_GALLERY = 'OPEN_PRODUCT_GALLERY';
 export const REQUEST_PRODUCT_DATA = 'REQUEST_PRODUCT_DATA';
 
 /**
  * Max number of reviews shown
  * @type {number}
  */
-export const REVIEW_PREVIEW_LIMIT = 20;
+export const REVIEW_PREVIEW_LIMIT = 2;
```
Comment (richardgorman) [inline pages/Product/constants.js]: Rename to `REVIEWS_PREVIEW_LIMIT`?

File: pages/Product/index.jsx [hunk 12]
```diff
@@ -4,17 +4,16 @@
  * This source code is licensed under the Apache 2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
-
 import React, { Component } from 'react';
```
Comment (richardgorman) [inline pages/Product/index.jsx]: Space after LICENCE block.

File: pages/Reviews/components/LoadMore/connector.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  getReviewsFetchingState,
+} from '@shopgate/pwa-common-commerce/reviews/selectors';
+import fetchReviews from '@shopgate/pwa-common-commerce/reviews/actions/fetchReviews';
+
+/**
+ * Maps the contents of the state to the component props.
+ * @param {Object} state The current application state.
+ * @return {Object} The extended component props.
+ */
+const mapStateToProps = state => ({
```
Comment (richardgorman) [inline pages/Reviews/components/LoadMore/connector.js]: Please order these props.

File: pages/Reviews/components/LoadMore/index.jsx [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+   * @type {Object}
+   */
+  static propTypes = {
+    fetchReviews: PropTypes.func.isRequired,
+    currentReviewCount: PropTypes.number,
+    isFetching: PropTypes.bool,
+    productId: PropTypes.string,
+    totalReviewCount: PropTypes.number,
+  };
+  /**
```
Comment (richardgorman) [inline pages/Reviews/components/LoadMore/index.jsx]: Missing line break.

File: pages/Reviews/index.jsx [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { hex2bin } from '@shopgate/pwa-common/helpers/data';
+import View from 'Components/View';
+import List from 'Components/Reviews/components/List';
+import LoadMoreButton from './components/LoadMore';
+import connect from './connector';
+import {
```
Comment (richardgorman) [inline pages/Reviews/index.jsx]: Please make this a single line as it's only importing one thing.

File: pages/Reviews/index.jsx [hunk 16]
```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { Component } from 'react';
```
Comment (richardgorman) [inline pages/Reviews/index.jsx]: Missing line break.

File: pages/Reviews/index.jsx [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+import { hex2bin } from '@shopgate/pwa-common/helpers/data';
+import View from 'Components/View';
+import List from 'Components/Reviews/components/List';
+import LoadMoreButton from './components/LoadMore';
+import connect from './connector';
+import {
+  REVIEW_ITEMS_PER_PAGE,
+} from './constants';
+/**
+ *
```
Comment (richardgorman) [inline pages/Reviews/index.jsx]: Empty block comment. Also missing space above.

File: pages/Reviews/index.jsx [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+    reviews: [],
+  };
+
+  /**
+   * Fetch the first reviews bundle.
+   */
+  componentDidMount() {
+    const { productId } = this.props.params;
+    // ProductId comes from url here, need to encode it first.
+    this.props.fetchReviews(hex2bin(productId), REVIEW_ITEMS_PER_PAGE, 0);
```
Comment (richardgorman) [inline pages/Reviews/index.jsx]: We want to move away from lifecycle hooks calling for data. Data fetching should now occur in the Observable Streams.

File: pages/Reviews/index.jsx [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  /**
+   * Gets a title translations.
+   * @returns {string} Title
+   */
+  getTitle() {
```
Comment (richardgorman) [inline pages/Reviews/index.jsx]: This could be refactored to be a getter.

File: pages/Reviews/index.jsx [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+ */
+class Reviews extends Component {
+  /**
+   * Context types definition.
+   * @type {{i18n: shim}}
+   */
+  static contextTypes = {
+    i18n: PropTypes.func,
+  };
+  /**
```
Comment (richardgorman) [inline pages/Reviews/index.jsx]: Missing line break.

File: pages/Reviews/mock.js [hunk 21]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {
```
Comment (richardgorman) [inline pages/Reviews/mock.js]: Missing line break.

File: components/Reviews/components/AllReviewsLink/style.js (hunk 1 above)
Comment (sznowicki) [reply components/Reviews/components/AllReviewsLink/style.js]: done

File: components/Reviews/components/AllReviewsLink/connector.js (hunk 2 above)
Comment (sznowicki) [reply components/Reviews/components/AllReviewsLink/connector.js]: done

File: components/Reviews/components/AllReviewsLink/index.jsx (hunk 3 above)
Comment (sznowicki) [reply components/Reviews/components/AllReviewsLink/index.jsx]: done

File: components/Reviews/components/AllReviewsLink/index.jsx (hunk 4 above)
Comment (sznowicki) [reply components/Reviews/components/AllReviewsLink/index.jsx]: done

File: components/Reviews/components/AllReviewsLink/index.jsx (hunk 5 above)
Comment (sznowicki) [reply components/Reviews/components/AllReviewsLink/index.jsx]: True. Adjusted.

File: components/Reviews/components/Header/components/RatingCount/spec.jsx (hunk 6 above)
Comment (sznowicki) [reply components/Reviews/components/Header/components/RatingCount/spec.jsx]: done

File: components/Reviews/mock.js (hunk 7 above)
Comment (sznowicki) [reply components/Reviews/mock.js]: done

File: components/Reviews/mock.js (hunk 8 above)
Comment (sznowicki) [reply components/Reviews/mock.js]: Renamed.

File: components/Reviews/spec.jsx (hunk 9 above)
Comment (sznowicki) [reply components/Reviews/spec.jsx]: done

File: components/View/mock.jsx (hunk 10 above)
Comment (sznowicki) [reply components/View/mock.jsx]: done

File: pages/Product/constants.js (hunk 11 above)
Comment (sznowicki) [reply pages/Product/constants.js]: As you wish.

File: pages/Product/index.jsx (hunk 12 above)
Comment (sznowicki) [reply pages/Product/index.jsx]: done

File: pages/Reviews/components/LoadMore/connector.js (hunk 13 above)
Comment (sznowicki) [reply pages/Reviews/components/LoadMore/connector.js]: done

File: pages/Reviews/components/LoadMore/index.jsx (hunk 14 above)
Comment (sznowicki) [reply pages/Reviews/components/LoadMore/index.jsx]: done

File: pages/Reviews/index.jsx (hunk 15 above)
Comment (sznowicki) [reply pages/Reviews/index.jsx]: done

File: pages/Reviews/index.jsx (hunk 16 above)
Comment (sznowicki) [reply pages/Reviews/index.jsx]: done

File: pages/Reviews/index.jsx (hunk 17 above)
Comment (sznowicki) [reply pages/Reviews/index.jsx]: done

File: pages/Reviews/index.jsx (hunk 19 above)
Comment (sznowicki) [reply pages/Reviews/index.jsx]: Changed it to a getter.

File: pages/Reviews/index.jsx (hunk 20 above)
Comment (sznowicki) [reply pages/Reviews/index.jsx]: done

File: pages/Reviews/mock.js (hunk 21 above)
Comment (sznowicki) [reply pages/Reviews/mock.js]: done

**Pull request merged**